src/bandersnatch/mirror.py

```diff
@@ -613,8 +616,8 @@ def json_pypi_symlink(self, package_name: str) -> Path:
 
     def simple_directory(self, package: Package) -> Path:
         if self.hash_index:
-            return Path(self.webdir / "simple" / package.name[0] / package.name)
-        return Path(self.webdir / "simple" / package.name)
+            return self.webdir / "simple" / package.name[0] / package.name
+        return self.webdir / "simple" / package.name
```

Comment:
I would like to maintain everything as Path() objects ... We should cast to str when ever you hand it off to the s3 libraries ...

Reply:
OK, I'll try to fix this issue by handling this case in storage_plugin